### PR TITLE
Add `Tablet/Trackpad` 3D navigation preset

### PR DESCRIPTION
**Pull request by RobProductions**
Following up on the discussion from #85331 , if a user enables "emulate 3 button mouse" with the `Godot` preset, they will see the camera always orbit because the `Godot` preset doesn't use any hotkey for orbit. Bringing in "Alt" as a fallback key for 3 button mouse users (which is what the code used to do) would eliminate their ability to customize the orbit button, so instead the idea of a `Tablet/Trackpad` preset seems to work much better.

With this change, 3 button mouse is now tied to the navigation preset, so if you set the preset to `Tablet/Trackpad` it will enable 3 button mouse and also set the orbit key to "Alt" so that Tablet users have better default settings. I also added options and functionality for mouse buttons 4 and 5; none of the presets use this but this should allow users to customize 3D nav further for specific tablet pens or mice.

image

With this, tablet and trackpad users can simply change the preset setting rather than changing "Emulate 3 button mouse" which should provide a clearer experience now that everything is customizable. Similarly, if someone changes "Emulate 3 button mouse" from the default the preset will change to `Custom` which should help clarify that they are going outside of the default options. 

~~One question I had about the presets: is it okay to keep pushing down the `Custom` preset to a different number if we add more presets that go above it in the UI? I numbered them in this commit for clarity but say for example we add a `Blender` preset in the future, is it okay for that to become element 4 with `Custom` getting pushed to element 5? Or should I just reorder `Custom` to be at the top now for better futureproofing?~~

Thanks and let me know if there are any questions :)

### CODE REVIEW

File: editor/editor_settings.cpp [hunk 1]
```diff
@@ -811,10 +811,10 @@ void EditorSettings::_load_defaults(Ref<ConfigFile> p_extra_config) {
 	// 3D: Navigation
 	_initial_set("editors/3d/navigation/invert_x_axis", false, true);
 	_initial_set("editors/3d/navigation/invert_y_axis", false, true);
-	EDITOR_SETTING_BASIC(Variant::INT, PROPERTY_HINT_ENUM, "editors/3d/navigation/navigation_scheme", 0, "Godot,Maya,Modo,Custom")
-	EDITOR_SETTING_BASIC(Variant::INT, PROPERTY_HINT_ENUM, "editors/3d/navigation/orbit_mouse_button", 1, "Left Mouse,Middle Mouse,Right Mouse")
-	EDITOR_SETTING_BASIC(Variant::INT, PROPERTY_HINT_ENUM, "editors/3d/navigation/pan_mouse_button", 1, "Left Mouse,Middle Mouse,Right Mouse")
-	EDITOR_SETTING_BASIC(Variant::INT, PROPERTY_HINT_ENUM, "editors/3d/navigation/zoom_mouse_button", 1, "Left Mouse,Middle Mouse,Right Mouse")
+	EDITOR_SETTING_BASIC(Variant::INT, PROPERTY_HINT_ENUM, "editors/3d/navigation/navigation_scheme", 0, "Godot,Maya,Modo,Tablet/Trackpad,Custom")
```
Comment (AThousandShips) [inline editor/editor_settings.cpp:814]: This breaks compatibility for anyone who has selected `Custom`

Comment (RobProductions) [reply editor/editor_settings.cpp:814]: Right, that's why I asked the question about futureproofing in the PR; is there a way to order the enum a specific way in editor settings without changing the int value of `Custom`?

**Comment by AThousandShips** [comment]
The new mode needs to be documented in the settings docs

File: editor/editor_settings.cpp (hunk 1 above)
Comment (AThousandShips) [reply editor/editor_settings.cpp:814]: I'm not sure, I'd say to put it after `Custom` in the values, but order in the same way like so:
```suggestion
	EDITOR_SETTING_BASIC(Variant::INT, PROPERTY_HINT_ENUM, "editors/3d/navigation/navigation_scheme", 0, "Godot:0,Maya:1,Modo:2,Tablet/Trackpad:4,Custom:3")
```

**Comment by RobProductions** [comment]
Thanks for the help! I've updated the docs and changed `Custom` back to value 3 so that shouldn't be a compat issue anymore, let me know if that looks good

File: doc/classes/EditorSettings.xml [hunk 2]
```diff
@@ -357,11 +357,12 @@
...
-			The navigation scheme preset to use in the 3D editor. Changing this setting will affect the mouse button and modifier controls used to navigate the 3D editor viewport.
+			The navigation scheme preset to use in the 3D editor. Changing this setting will affect the mouse button, [member editors/3d/navigation/emulate_3_button_mouse] setting, and modifier keys used to navigate the 3D editor viewport.
 			All schemes can use [kbd]Mouse wheel[/kbd] to zoom.
 			- [b]Godot:[/b] [kbd]Middle mouse button[/kbd] to orbit. [kbd]Shift + Middle mouse button[/kbd] to pan. [kbd]Ctrl + Shift + Middle mouse button[/kbd] to zoom.
 			- [b]Maya:[/b] [kbd]Alt + Left mouse button[/kbd] to orbit. [kbd]Middle mouse button[/kbd] to pan, [kbd]Shift + Middle mouse button[/kbd] to pan 10 times faster. [kbd]Alt + Right mouse button[/kbd] to zoom.
 			- [b]Modo:[/b] [kbd]Alt + Left mouse button[/kbd] to orbit. [kbd]Alt + Shift + Left mouse button[/kbd] to pan. [kbd]Ctrl + Alt + Left mouse button[/kbd] to zoom.
+			- [b]Tablet/Trackpad:[/b] [kbd]Alt[/kbd] to orbit. [kbd]Shift[/kbd] to pan. [kbd]Ctrl[/kbd] to zoom. The [member editors/3d/navigation/emulate_3_button_mouse] setting will be set to true.
```
Comment (Mickeon) [inline doc/classes/EditorSettings.xml:365]:
```suggestion
			- [b]Tablet/Trackpad:[/b] [kbd]Alt[/kbd] to orbit. [kbd]Shift[/kbd] to pan. [kbd]Ctrl[/kbd] to zoom, enables 3-button mouse emulation mode.
```

File: doc/classes/EditorSettings.xml [hunk 3]
```diff
@@ -357,11 +357,12 @@
 			If [code]true[/code], invert the vertical mouse axis when panning, orbiting, or using freelook mode in the 3D editor.
 		</member>
 		<member name="editors/3d/navigation/navigation_scheme" type="int" setter="" getter="">
-			The navigation scheme preset to use in the 3D editor. Changing this setting will affect the mouse button and modifier controls used to navigate the 3D editor viewport.
+			The navigation scheme preset to use in the 3D editor. Changing this setting will affect the mouse button, [member editors/3d/navigation/emulate_3_button_mouse] setting, and modifier keys used to navigate the 3D editor viewport.
```
Comment (Mickeon) [inline doc/classes/EditorSettings.xml:360]:
```suggestion
			The navigation scheme preset to use in the 3D editor. Changing this setting will affect the mouse button and modifier keys used to navigate the 3D editor viewport.
```

File: doc/classes/EditorSettings.xml [hunk 4]
```diff
@@ -357,11 +357,12 @@
...
 			All schemes can use [kbd]Mouse wheel[/kbd] to zoom.
 			- [b]Godot:[/b] [kbd]Middle mouse button[/kbd] to orbit. [kbd]Shift + Middle mouse button[/kbd] to pan. [kbd]Ctrl + Shift + Middle mouse button[/kbd] to zoom.
 			- [b]Maya:[/b] [kbd]Alt + Left mouse button[/kbd] to orbit. [kbd]Middle mouse button[/kbd] to pan, [kbd]Shift + Middle mouse button[/kbd] to pan 10 times faster. [kbd]Alt + Right mouse button[/kbd] to zoom.
 			- [b]Modo:[/b] [kbd]Alt + Left mouse button[/kbd] to orbit. [kbd]Alt + Shift + Left mouse button[/kbd] to pan. [kbd]Ctrl + Alt + Left mouse button[/kbd] to zoom.
+			- [b]Tablet/Trackpad:[/b] [kbd]Alt[/kbd] to orbit. [kbd]Shift[/kbd] to pan. [kbd]Ctrl[/kbd] to zoom. The [member editors/3d/navigation/emulate_3_button_mouse] setting will be set to true.
 			See also [member editors/3d/navigation/orbit_mouse_button], [member editors/3d/navigation/pan_mouse_button], [member editors/3d/navigation/zoom_mouse_button], and [member editors/3d/freelook/freelook_navigation_scheme].
```
Comment (Mickeon) [inline doc/classes/EditorSettings.xml:366]:
```suggestion
			See also [member editors/3d/navigation/orbit_mouse_button], [member editors/3d/navigation/pan_mouse_button], [member editors/3d/navigation/zoom_mouse_button], and [member editors/3d/freelook/freelook_navigation_scheme], and [member editors/3d/navigation/emulate_3_button_mouse].
```

**Review by Mickeon — COMMENTED** [review]
I don't think it's worth being so explicit over this one editor setting. The other affected ones are listed at the bottom of the description.

**Review by Calinou — APPROVED** [review]
Tested locally on a MacBook Pro, it works as expected. The shortcuts with this preset make sense to me.

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**